fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    this.getAllTodos();
+  }
+
+  getAllTodos() {
+    fetch('/api/todos')
+      .then(res => res.json())
+      .then(todos => this.setState({
+        todos: todos
+      }))
+      .catch(err => console.error(err));
+  }
+
+  addTodo(newTodo) {
+    fetch('/api/todos', {
+      method: 'POST',
+      body: JSON.stringify(newTodo),
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    })
+      .then(res => res.json())
+      .then(todo => this.setState(
+        state => {
+          state.todos.push(todo);
+          return { todos: state.todos };
```

Comment:
In react we always replace our state properties, so reference data (like objects and arrays) need to be copied before mutation. Alternatively, `Array.prototype.concat()` returns a new array without modifying the original .

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat